doc/contributing/docs/style.rst

```diff
@@ -174,13 +174,38 @@ Spelling and punctuation
 US vs British spelling
 ~~~~~~~~~~~~~~~~~~~~~~
 
-We use the US English spelling.
+Use the US English spelling.
 
 Check your spelling and punctuation
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Consider checking spelling, grammar, and punctuation with special tools.
 
+Dashes
+~~~~~~
+
+There is no need to insert special symbols like dashes, quotation marks, or apostrophes.
+The documentation :doc:`builder </contributing/docs/build/>`
+renders specific character sequences in the source into correct typographic characters. 
+
+Type three hyphens (``---``) to insert an em dash (---)
+and two hyphens (``--``) for an en dash (--).
+The following rules apply:
+
+*   The shorter **en dash** ``--`` is used to mark ranges and with the meaning "to"
+    in phrases like "Dover--Calais crossing".
+*   The longer **em dash** ``---`` is used to separate extra information or mark a break in a sentence.
+
+Don't use a single hyphen as a dash. Don't add spaces on either side of a dash. 
+
+When indicating a range like ``code element 1``\--``code element 2``, escape the series of hyphens using
+`character-level inline markup <https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#character-level-inline-markup-1>`_:
+
+..  code-block:: rst
+
+    ``box.begin()``\--``box.commit()``
```

Comment:
what's the use case for this syntax? range between two functions?

Reply:
This is a real example:
![image](https://user-images.githubusercontent.com/40490252/150929235-e6a6947d-720f-42d7-8f8e-79fc7eeeb434.png)